DQM/SiStripCommon/src/TkHistoMap.cc

```diff
@@ -78,27 +85,43 @@ void TkHistoMap::loadTkHistoMap(std::string path, std::string MapName, bool mech
 }
 
 void TkHistoMap::createTkHistoMap(std::string& path, std::string& MapName, float& baseline, bool mechanicalView){
-  
+
   int nchX;
   int nchY;
   double lowX,highX;
   double lowY, highY;
   std::string fullName, folder;
 
-  tkHistoMap_.resize(HistoNumber);    
+  const bool bookTH2F = isTH2F_;
+  tkHistoMap_.resize(HistoNumber);
   for(int layer=1;layer<HistoNumber;++layer){
     folder=folderDefinition(path,MapName,layer,mechanicalView,fullName);
     tkdetmap_->getComponents(layer,nchX,lowX,highX,nchY,lowY,highY);
-    MonitorElement* me  = dqmStore_->bookProfile2D(fullName.c_str(),fullName.c_str(),
+    MonitorElement* me;
+
+    if(bookTH2F==false){
+
+    me  = dqmStore_->bookProfile2D(fullName.c_str(),fullName.c_str(),
```

Comment:
>  The one in line 104 is not so trivial to remove since there is no ibooker in the function.

This is part of the problem. DQMStore is a 'global' object, so any booking must acquire a lock in order to properly book and not crash/corrupt the store.
The provided bookHistograms and few other functions (which provide IBooker...) automatically acquire this lock.